registry/handlers/referrers.go

```diff
@@ -76,10 +98,20 @@ func (h *referrersHandler) GetReferrers(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	if referrers == nil {
+	startIndex := pageNumber * pageSize
+
+	if referrers == nil || startIndex > len(referrers) {
 		referrers = []v1.Descriptor{}
 	}
```

Comment:
When fall into this code path, the code from line 107 - 113 is still executed but should not.

Reply:
Added back the if check if `len(referrers) > pageSize` to enclose this part.